public/JS/createChat.js

```diff
@@ -0,0 +1,101 @@
+$(function () {
+/*--------------------------------------------------- Click events ---------------------------------------------------*/
+  // Button that directs user from chat create to lobby screen and show updated list of chats
+  $('#createChatToLobby').click(function(){
+    $('#lobby').show();
+    $('#chatCreate').hide();
+    updateChats();
+  })
+
+  // Button that directs user from chat create to home screen
+  $('#createChatToHomeButton').click(function(){
+    $('#chatCreate').hide();
+    $('#home').show();
+  })
+
+  // Button that requests creating a chat
+  $('#submitChat').click(function(){
+    //Check if user has selected themselves or not. We cannot allow a user to create a chat in which they are not participating
+    if($('#modDisplay').val().includes(displayName) || $('#usersDisplay').val().includes(displayName)){
+      socket.emit('create chat', {users: $('#usersDisplay').val(), mods: $('#modDisplay').val(), chatname: $('#chatname').val()});
+      $('#chatname').val('');    //clear the chat name text box
+    }
+    else alert("Please include yourself as a participant in this chat");
+  })
+
+   //Button that rejects chat creation request
+   $('#rejectChatButton').click(function(){
+    var chatsDisplayed = document.getElementById("chatsDisplayed")
+    var chat = chatsDisplayed.selectedIndex;
+    var name=chatsDisplayed.children[chat].innerHTML
+    socket.emit("rejectChat",name)
+    chatsDisplayed.options.remove(chat);
+  })  
+
+  //Button that approves chat creation request
+  $('#approveChatButton').click(function(){
+    var chatsDisplayed = document.getElementById("chatsDisplayed")
+    var chat = chatsDisplayed.selectedIndex;
+    var name=chatsDisplayed.children[chat].innerHTML
+    socket.emit("approveChat",name)
+    alert(name + " has been approved");
+  })
+
+/*--------------------------------------------------- Socket.on events ---------------------------------------------------*/
+  // TODO: I don't know what this does. Needs more comments
```

Comment:
This is how the "create chat" screen is populated. It first clears the boxes for mods and users. Then it uses the list of mods and users inside of userObj to display them in their respective boxes for users to select.

Reply:
That makes more sense. I updated the above comment